### DISCUSSION TITLE
failed to copy linked module directory to layer path:

### DISCUSSION THREAD

**Discussion by mishageorge0** [paketo-buildpacks/feedback#322] [Builder Team]
while building image in docker, I am getting error.
 
failed to copy linked module directory to layer path: failed to copy: destination exists: remove /layers/paketo-buildpacks_npm-install/launch-modules: directory not empty
ERROR: failed to build: exit status 1
ERROR: failed to build: executing lifecycle: failed with status code: 51
 
This issue is inconsistent, and I have tried many workarounds, but they didn’t help.
pack build misha2012/incident-management-srv:latest \
    --path gen/srv \
    --builder paketobuildpacks/builder-jammy-base \
    --publish

**Comment by dmikusa** [comment]
There's not a lot to go on here. It looks like lifecycle is having a problem.

Here are a couple of very general suggestions:

- Try `pack build` with the `--clear-cache` option which would clear out cache data. Maybe something is in a weird state there. Alternatively, change `misha2012/incident-management-srv:latest` to something else. That has the same effect. New image name, all new caches. 
- Try w/out the publish flag and see if you get the same issue. Publish behaves differently than building in a local Docker daemon. If it suddenly works, that would narrow it down to something problematic with that flag.
- If you're using PodMan, that sometimes has issues. If you have access to a system with Docker, give that a try and see if it works for you.

If you can post more information, like the full build output with the `pack build -v` flag enabled, that might help debug further.

Hope that helps!
